app-modules/group/src/Policies/GroupPolicy.php

```diff
@@ -60,83 +60,83 @@ public function viewAny(Authenticatable $authenticatable): Response
+            return Response::deny('You do not have permission to view this group.');
         }
 
         return $authenticatable->canOrElse(
-            abilities: ["segment.{$segment->getKey()}.view"],
-            denyResponse: 'You do not have permission to view this segment.'
+            abilities: ["segment.{$group->getKey()}.view"],
+            denyResponse: 'You do not have permission to view this group.'
         );
     }
 
     public function create(Authenticatable $authenticatable): Response
     {
         return $authenticatable->canOrElse(
             abilities: 'segment.create',
-            denyResponse: 'You do not have permission to create segments.'
+            denyResponse: 'You do not have permission to create groups.'
         );
     }
 
-    public function update(Authenticatable $authenticatable, Segment $segment): Response
+    public function update(Authenticatable $authenticatable, Group $group): Response
     {
-        if (! $authenticatable->hasLicense($segment->model?->class()::getLicenseType())) {
-            return Response::deny('You do not have permission to update this segment.');
+        if (! $authenticatable->hasLicense($group->model->class()::getLicenseType())) {
+            return Response::deny('You do not have permission to update this group.');
         }
 
         return $authenticatable->canOrElse(
-            abilities: ["segment.{$segment->getKey()}.update"],
-            denyResponse: 'You do not have permission to update this segment.'
+            abilities: ["segment.{$group->getKey()}.update"],
+            denyResponse: 'You do not have permission to update this group.'
         );
     }
 
-    public function delete(Authenticatable $authenticatable, Segment $segment): Response
+    public function delete(Authenticatable $authenticatable, Group $group): Response
     {
-        if (! $authenticatable->hasLicense($segment->model?->class()::getLicenseType())) {
-            return Response::deny('You do not have permission to delete this segment.');
+        if (! $authenticatable->hasLicense($group->model->class()::getLicenseType())) {
+            return Response::deny('You do not have permission to delete this group.');
         }
 
-        if ($segment->campaigns()->exists()) {
-            return Response::deny('This population segment is associated with a campaign. Please remove the campaign before attempting to remove the segment.');
+        if ($group->campaigns()->exists()) {
+            return Response::deny('This population group is associated with a campaign. Please remove the campaign before attempting to remove the group.');
         }
 
         return $authenticatable->canOrElse(
-            abilities: ["segment.{$segment->getKey()}.delete"],
-            denyResponse: 'You do not have permission to delete this segment.'
+            abilities: ["segment.{$group->getKey()}.delete"],
```

Comment:
Since we are in this file now, please change this to the `*` version of the permission check, don't pass the key